console/console-init/ui/src/modules/device/components/DeviceFilterCriteria/DeviceFilterCriteria.tsx

```diff
@@ -43,37 +48,84 @@ const DeviceFilterCriteria: React.FunctionComponent<IDeviceFilterCriteriaProps>
   deleteCriteria,
   setCriteria
 }) => {
-  const onChangeParameter = (value: string) => {
-    const criteriaObj = criteria;
-    criteria.parameter = value;
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+  const options: ISelectOption[] = [
+    {
+      value: "accelerated speed",
+      label: "accelerated speed",
+      key: "accelerated speed"
+    },
+    { value: "distance", label: "distance", key: "distance" },
+    { value: "humidity", label: "humidity", key: "humidity" },
+    { value: "item", label: "item", key: "item" },
+    { value: "location", label: "location", key: "location" },
+    { value: "latitude", label: "latitude", key: "latitude" },
+    { value: "temparature", label: "temparature", key: "temparature" },
+    { value: "team", label: "team", key: "team" },
+    { value: "template", label: "template", key: "template" },
+    { value: "tend", label: "tend", key: "tend" }
+  ];
+  const onChangeParameter = (event: any) => {
+    const value = event.target.value;
+    const criteriaObj = { ...criteria };
+    criteriaObj.parameter = value;
     setCriteria(criteriaObj);
+    const optionList: ReactElement[] = [];
+    return optionList;
   };
 
   const onChangeValue = (value: string) => {
-    const criteriaObj = criteria;
-    criteria.value = value.trim();
+    const criteriaObj = { ...criteria };
+    criteriaObj.value = value.trim();
     setCriteria(criteriaObj);
   };
 
   const onSelectOperator = (value: string) => {
-    const criteriaObj = criteria;
-    criteria.operator = value;
+    const criteriaObj = { ...criteria };
+    criteriaObj.operator = value;
     setCriteria(criteriaObj);
   };
 
+  const onClearParameter = () => {
+    const criteriaObj = { ...criteria };
+    criteriaObj.parameter = "";
+    setCriteria(criteriaObj);
+  };
+
+  const onSelect = (event: any, value: string | SelectOptionObject) => {
+    const criteriaObj = { ...criteria };
+    criteriaObj.parameter = value.toString();
```

Comment:
why do we need toString, value is already string. value type is string that will handle it.

Reply:
value type may be SelectOptionObject .